aiohttp/connector.py

```diff
@@ -1160,6 +1199,10 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
                     if not self._cleanup_closed_disabled:
                         self._cleanup_closed_transports.append(transp)
                     last_exc = exc
+                    # Remove the bad peer from the list of addr_infos
+                    sock: socket.socket = transp.get_extra_info("socket")
+                    bad_peer = sock.getpeername()
+                    aiohappyeyeballs.remove_addr_infos(addr_infos, bad_peer[0])
```

Comment:
This should take the whole peer since it could mean we have different ports even if it's unlikely

Reply:
https://bugs.python.org/issue42620